### PR TITLE
chore(versions): Improve NavButton click handling and forward refs

### PR DESCRIPTION
**Pull request by jstoffan**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @jstoffan has signed the CLA. Thanks for the pull request!

**Comment by jstoffan** [comment]
I ended up needing the keypress logic and tests elsewhere, but the other changes in this pull request are still valuable.

File: src/elements/common/nav-button/NavButton.js [hunk 1]
```diff
@@ -21,18 +21,23 @@ type Props = {
...
-    className = 'bdl-NavButton',
-    component: Component = PlainButton,
-    exact,
-    onClick,
-    replace,
-    strict,
-    to,
-    ...rest
-}: Props) => {
+const isClickEvent = event => {
```
Comment (priyajeet) [inline src/elements/common/nav-button/NavButton.js]: Maybe this go inside utils/dom.js or utils/events.js or utils/keys.js?

File: src/elements/common/nav-button/__tests__/NavButton-test.js [hunk 2]
```diff
@@ -173,4 +173,61 @@ describe('elements/common/nav-button/NavButton', () => {
...
     });
+
+    describe('onClick', () => {
+        const mockHistory = {
+            listen: jest.fn(),
+            location: {},
+            push: jest.fn(),
+        };
+
+        it('calls onClick eventhandler and history.push', () => {
```
Comment (priyajeet) [inline src/elements/common/nav-button/__tests__/NavButton-test.js]: test() 🐙

File: src/elements/common/nav-button/NavButton.js (hunk 1 above)
Comment (priyajeet) [reply src/elements/common/nav-button/NavButton.js]: Also perhaps named `isLeftClickEvent()` since we are ignoring right clicks.

**Pull request merged**